cachito/web/auth.py

```diff
@@ -18,14 +18,45 @@ def user_loader(username):
     return User.query.filter_by(username=username).first()
 
 
+def _get_kerberos_principal(request):
+    """
+    Get the Kerberos principal from the current request.
+
+    This relies on the "REMOTE_USER" environment variable being set. This is usually set by the
+    mod_auth_gssapi Apache authentication module.
+
+    :param flask.Request request: the Flask request
+    :return: the user's Kerberos principal or None
+    :rtype: str
+    """
+    return request.environ.get('REMOTE_USER')
+
+
+def _get_cert_dn(request):
+    """
+    Get the client certificate's subject's distinguished name.
+
+    This relies on the "SSL_CLIENT_S_DN" environment variable being set. This is set by the mod_ssl
+    Apache module. If Apache is unable to verify the client certificate, no user will be returned.
+
+    :param flask.Request request: the Flask request
+    :return: the client certificate's subject's distinguished name or None
+    :rtype: str
+    """
+    ssl_client_verify = request.environ.get('SSL_CLIENT_VERIFY')
```

Comment:
Does this ensure the client cert has been issued by a trusted CA? Can we specify which CA should be used?

Reply:
This is answered by the fact that you're specifying the CA in the playbooks.